torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -2205,7 +2205,10 @@ at::Tensor XLANativeFunctions::nan_to_num(const at::Tensor& self,
   if (!at::native::is_floating_point(self)) {
     return CopyTensor(self);
   }
-  auto element_type = TensorTypeToRawXlaType(self.scalar_type());
+  XLATensor input_tensor = bridge::GetXlaTensor(self);
+  const Device& device = input_tensor.GetDevice();
+  auto element_type = GetDevicePrimitiveType(
```

Comment:
you can use `MakeXlaPrimitiveType` which takes scalar_type directly.

Reply:
Good shortcut. Thanks!